File: employeeManagementApp.Domain/Entities/Employee.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace employeeManagementApp.Domain.Entities
{
    public  class Employee
    {
        public string FirstName { get; set; }
        public string MiddleName { get; set; }
        public string LastName { get; set; }
        public DateTime BirthDate { get; set; }
        public DateTime HireDate { get; set; }
        public string Department { get; set; }
        public string JobTitle { get; set; }
        public int Salary { get; set; }
        [Key]
        [Required]
        public int? EmployeeId { get; set; }
        public string AvailableHours { get; set; }
       // public int salaryType { get; set; }
    }
}

File: employeeManagementApp.web/Scripts/DMACC/employee.js
function searchEmployee() {
    var search = $("#searchString").val();
    $.ajax({
        url: "Search",
        data: { searchString: search }
    }).done(function (data) {
        $("#FirstName").val(data.FirstName);
        $("#MiddleName").val(data.MiddleName);
        $("#LastName").val(data.LastName);
        $("#BirthDate").val(data.Birthdate);
        $("#HireDate").val(data.HireDate);
        $("#Department").val(data.Department);
        $("#JobTitle").val(data.JobTitle);
        $("#Salary").val(data.Salary);
        $("#EmployeeId").val(data.EmployeeId);
        $("#AvailableHrs").val(data.AvailableHrs);
    });
}

function updateEmployee() {
    var firstName = $("#FirstName").val();
    var middleName = $("#MiddleName").val();
    var lastName = $("#LastName").val();
    var birthDate = $("#BirthDate").val();
    var hireDate = $("#HireDate").val();
    var department = $("#Department").val();
    var jobTitle = $("#JobTitle").val();
    var salary = $("#Salary").val();
    var employeeId = $("#EmployeeId").val();
    var availableHrs = $("#AvailableHrs").val();

    //window.alert(firstName + "," + middleName + "," + lastName + "," + birthDate);
    $.ajax({
        url: "UpdateEmployee",
        dataType: "Json",
        data: {
            FirstName: firstName,
            MiddleName: middleName,
            LastName: lastName,
            BirthDate: birthDate,
            HireDate: hireDate,
            Department: department,
            JobTitle: jobTitle,
            Salary: salary,
            EmployeeId: employeeId,
            AvailableHrs: availableHrs
        }
    }).done(function(data) {
        if (data) {
            $("#successMessage").removeClass("hidden")
                .addClass("visible");
        } else {
            $("#errorMessage").removeClass("hidden")
                .addClass("visible");
        }
    });
}

 